MetaMask/metamask-mobile
# chore: added new privacy file and identified reason for using privacy APIs

## **Description**

This PR adds a new privacy file identifying why the app is using specific privacy APIs. 

These mentions in the manifest:
- NSPrivacyAccessedAPICategoryFileTimestamp - Display Activity Times
- NSPrivacyAccessedAPICategoryDiskSpace - Download files
- NSPrivacyAccessedAPICategorySystemBootTime - Sentry Times
- NSPrivacyAccessedAPICategoryUserDefaults - MMKV
Are the modules/functions that are associated with the required declarations in the manifest. 

Sentry was also updated due to bug with duplicate [Privacy Manifests in older versions](https://github.com/getsentry/sentry-react-native/issues/3599).

## **Related issues**

[Fixes](https://github.com/MetaMask/mobile-planning/issues/1424)

## **Manual testing steps**

1. Submit new app to TestFlight
2. See there are now errors in privacy audit

## **Screenshots/Recordings**

### **Before**

<img width="882" alt="Screenshot 2024-04-05 at 12 39 25 PM" src="https://github.com/MetaMask/metamask-mobile/assets/10342624/56b59718-366e-4365-b854-5cd120d07544">

### **After**

<img width="744" alt="Screenshot 2024-04-04 at 12 31 53 PM" src="https://github.com/MetaMask/metamask-mobile/assets/10342624/3a0e9a15-624f-4fb6-8050-ecdb5110523f">


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
